# 복습

"파이썬은 괄호 대신에 들여쓰기로 블럭(실행영역)을 구분한다."

예:


```
for i in range(1,11):  
    print('a')   # <-- for loop 문의 관할하에 있는 실행영역
print('b') # <-- for loop 문의 관할하에 있는 실행영역이 아닙니다.
           # for loop 문이 다 끝나고 나면 작동되는 문장.
```

# SQL과 파이썬의 차이점?

1. SQL은 비절차적 언어
```
select ename, sal
    from emp
    where ename = 'scott'
```




# 26. 시퀀스 자료형 이해하기

시퀀스 자료형은



설명: 문자형 변수에서 특정 철자만 가져오고 싶으면 문자형 변수 옆에 대괄호를 쓰고 순서에 해당하는 숫자를 적어주면 됩니다.

문제88. 아래의 문자형 변수에서 맨 끝의 철자인 h를 출력하시오!


In [37]:
b = 'smith'
print(b[4])                  # s  m  i  t  h
print(b[-1])                 # 0  1  2  3  4
print(b[len(b)-1])           #-5 -4 -3 -2 -1

h
h
h


문제 89. 판다스를 이용해서 emp3.scv에서 이름을 출력하는데 이름의 첫번째 철자만 출력하시오!


In [18]:
# SQL
# select substr(ename,1,1)
#   from emp;
filename = '/content/emp3.csv'
import pandas as pd
emp = pd.read_csv(filename)
for i in emp['ename']:
    print(i[0])


K
B
C
J
M
A
T
J
W
F
S
S
A
M


자료형을 확인해보자  
설명: type 명령어는 변수가 문자형, 숫자형, 리스트형, 딕셔너리형, 튜플형인지 확인하는 함수입니다.

In [20]:
for i in emp['ename']:
    print(type(i))

<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>


series는 오라클로 치면 컬럼이다.

In [23]:
print(type(emp['ename']))

<class 'pandas.core.series.Series'>


설명 : <class 'str'> 문자형으로 변경해줘야 문자형에서 특정 요소를 가져오는 문법을 사용할 수 있습니다.  
  
예:  
```a[0]```

판다스의 <class 'pandas.core.series.Series'> 를 문자형으로 변경하려면 for loop 문을 이용해서 하나씩 가져오면 됩니다.

예:
```
for i in emp['ename']:
    print (i)
```

문제90. 판다스를 이용해서 emp3.csv를 가져와서 이름의 끝글자를 출력하시오.  

In [25]:
import pandas as pd
filename = '/content/emp3.csv'
emp = pd.read_csv(filename)
 
for i in emp['ename']:
    print(i[-1])

G
E
K
S
N
N
R
S
D
D
H
T
S
R


문제91. 위의 결과에서 앞에 이름도 같이 출력하시오!

In [29]:
import pandas as pd
filename = '/content/emp3.csv'
emp = pd.read_csv(filename)
for i in emp['ename']:
    print(i, '  ', i[-1])

KING    G
BLAKE    E
CLARK    K
JONES    S
MARTIN    N
ALLEN    N
TURNER    R
JAMES    S
WARD    D
FORD    D
SMITH    H
SCOTT    T
ADAMS    S
MILLER    R


문제92. 이름의 첫번째 철자가 S로 시작하는 사원들의 이름을 출력하시오.

In [32]:
# SQL
# select ename
#   from emp
#   where ename like 'S%';

import pandas as pd
filename = '/content/emp3.csv'

emp = pd.read_csv(filename)
for i in emp['ename']:
  if i[0] == 'S':
    print(i)

SMITH
SCOTT


문제93. 이름의 끝 글자가 T로 끝나는 사원들의 이름을 출력하시오.

In [35]:
# SQL
# select ename
#   from emp
#   where ename like '%T';

import pandas as pd
filename ='/content/emp3.csv'

emp = pd.read_csv(filename)
for i in emp['ename']:
  if i[-1] == 'T':
    print(i)

SCOTT


# 27. 시퀀스 자료 인덱싱 이해하기

인덱싱(indexing) 이란 시퀀스 자료형에서 인덱스를 통해 해당하는 값을 얻는 방법입니다. 파이썬에서는 인덱스를 0부터 시작하며 음수인 인덱스도 사용가능합니다. 음수 인덱스는 '끝에서 몇 번째' 라는 의미를 갖습니다.

예:
```
a = 'SALESMAN'
print( a[2] ) # L이 출력됨
```
  

문제94. 이름의 두번째 철자가 M 인 사원의 이름을 출력하시오.

In [42]:
# SQL
# select ename, sal
#   from emp
#   where ename like = '_M%'

for i in emp['ename']:
  if i[1] == 'M':
    print(i)

SMITH


# 28. 시퀀스 자료 슬라이싱 이해하기

인덱싱은 인덱스에 해당하는 요소 하나를 취하는 방법이지만 슬라이싱은 시퀀스 자료에서 일정 범위에 해당하는 부분을 취하는 방법입니다.

예제:  


In [43]:
a = 'scott'
print( a[1:3] )

co


설명: 1번째 자리부터 3 미만의 자리까지 요소들을 출력  
```
s c o t t
0 1 2 3 4


문제 95. 아래의 SQL을 판다스로 구현하시오!  
SQL> select substr(ename,1,3)  
      from emp;


In [46]:
for i in emp['ename']:
  print(i[0:3])

KIN
BLA
CLA
JON
MAR
ALL
TUR
JAM
WAR
FOR
SMI
SCO
ADA
MIL


문제96. 아래의 SQL을 판다스로 구현하시오!  
SQL>  
select substr(ename, -2, 2)
from emp

In [102]:
for i in emp['ename']:
  print(i[-2:])

NG
KE
RK
ES
IN
EN
ER
ES
RD
RD
TH
TT
MS
ER


-2: 이거는 -2부터 끝까지 읽으라는 말이다.  
-2:-1을 하게되면 끝 글자가 안나온다.  
.  
문제97.(점심시간 문제) 아래의 SQL을 판다스로 구현하시오!  
>SQL  
select substr(ename,2,3)  
from emp;

In [105]:
for i in emp['ename']:
  print(i[1:4])

ING
LAK
LAR
ONE
ART
LLE
URN
AME
ARD
ORD
MIT
COT
DAM
ILL


# 29. 시퀀스 자료 연결 이해하기(+)

자료형이 동일한 두 개의 시퀀스 자료는 + 연산자로 순서있게 연결하여 새로운 시퀀스 자료를 만들 수 있습니다.  
문자열 + 문자열, 리스트 + 리스트, 튜플 + 튜플과 같이 두개의 동일한 시퀀스 자료형에 대해 '+' 연산자로 연결이 가능합니다.


예제 :

In [106]:
a = 'I love '
b = 'python'
print( a + b)

love python


문제98. 아래의 두개의 리스트를 연결하여 아래와 같이 출력하시오!



In [107]:
a = [2,3,4,5]
b = [9,2,4,8]
print(a + b)

[2, 3, 4, 5, 9, 2, 4, 8]


오라클은 ||인데 파이썬은 + 이다.
아래의 SQL을 판다스로 구현하시오!

SQL> select ename || sal
  from emp

  

In [127]:
for i, k in zip(emp['ename'], emp['sal']):
  print (i + str(k))

KING5000
BLAKE2850
CLARK2450
JONES2975
MARTIN1250
ALLEN1600
TURNER1500
JAMES950
WARD1250
FORD3000
SMITH800
SCOTT3000
ADAMS1100
MILLER1300


설명: zip 함수를 for loop문에 사용하게 되면 두 개의 범위 데이터 한번에 받아서 loop를 수행할 수 있습니다. str 함수를 사용한 이유는 문자형 + 문자형을 연결이 가능한데 문자형 + 숫자형은 연결이 안됩니다. 그래서 숫자형을 문다형으로 변환하기 위해서 str 함수를 사용했습니다.

# 30. 시퀀스 자료 반복 이해하기(*)

동일한 시퀀스 자료를 반복하여 새로운 시퀀스 자료로 만들고자 하면 별표(*)를 연산자로 사용합니다.  
예제:

In [128]:
print('a' * 7)  # 소문자 알파벳 a를 7번 출력합니다.

aaaaaaa


In [129]:
print([1,2,3] * 5) # 리스트 [1,2,3]안의 요소를 5개로 늘려서 출력

[1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3]


문제100. 주사위의 눈 6개를 100개 담은 리스트 dice100을 만들고 dice100 리스트를 출력하시오!

In [132]:
dice100 = [1,2,3,4,5,6]* 100
print(dice100)

[1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 

문제101. 초등학생 키가 10개가 들어있는 아래의 tall 리스트의 요소 10개를 10000개로 증가시켜서 tall10000 리스트에 담고 출력하시오.

In [136]:
tall = [129.3,130.2,132.5,134.7,136.3,137.8,138.1,140.2,142.3,145.2]
tall10000=tall * 10000
print(tall10000)              # <class 'list'>

[129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5, 134.7, 136.3, 137.8, 138.1, 140.2, 142.3, 145.2, 129.3, 130.2, 132.5

문제102. 위의 모집단의 평균값을 출력하시오!

In [140]:
import numpy as np
print(np.mean(tall10000))
print(np.var(tall10000))
print(np.std(tall10000))

136.66
23.782399999999985
4.876720209321013


설명: np.mean(리스트)를 사용하면 평균값이 출력됩니다.  
.  
문제103. 위의 모집단 tall10000에서 표본을 20개를 랜덤으로 추출하시오!

In [145]:
import random
a = []
for i in range(1,21):                   # 20번 반복하면서
  print(random.choice(tall10000))    # 모집단에서 랜덤으로 키하나를 추출

137.8
134.7
130.2
129.3
129.3
140.2
138.1
140.2
136.3
137.8
134.7
142.3
137.8
145.2
136.3
138.1
134.7
132.5
137.8
140.2


문제104. 위의 랜덤으로 추출한 표본 20개를 비어있는 a 리스트에 담으시오.

In [146]:
import random
a = []
for i in range(1,21):                   # 20번 반복하면서
  a.append(random.choice(tall10000))    # 모집단에서 랜덤으로 키하나를 추출
print(a)

[137.8, 140.2, 145.2, 134.7, 130.2, 140.2, 132.5, 145.2, 138.1, 137.8, 130.2, 136.3, 145.2, 129.3, 129.3, 136.3, 137.8, 145.2, 145.2, 132.5]


문제105. 위의 표본의 평균값을 출력하시오

In [151]:
print(np.mean(a))

137.45999999999998


# 31. 시퀀스 자료 크기 이해하기(len)

모든 시퀀스 자료는 고정된 길이 또는 크기를 가지고 있습니다.  
시퀀스 자료의 크기는 시퀀스 자료를 구성하는 요소의 갯수입니다.  
예제:  

In [152]:
a = 'scott'
print(len(a)) #5

b = [2,3,4,8,1]
print(len(b)) #리스트이ㅡ 요소의 갯수가 출력됩니다.

5
5


문제106. 아래의 SQL을 판다스로 구현하시오.  
SQL>  
select ename, length(ename)  
from emp;                         # 너무 어렵다...

In [187]:
for i in emp['ename']:
  print(i, len(i))

KING 4
BLAKE 5
CLARK 5
JONES 5
MARTIN 6
ALLEN 5
TURNER 6
JAMES 5
WARD 4
FORD 4
SMITH 5
SCOTT 5
ADAMS 5
MILLER 6


쉬운건데 SQL식으로 생각하다 보니까 자꾸 범위를 바꾸려고 한다. print 부분을 고쳐야한다.

numpy에서 그냥 var을 하면 표본의 분산에서 n-1을 해주지 않으므로 n/n-1을 곱해주어야 한다.  



# 32. 멤버체크 이해하기(in)

in은 자료에 어떤 값이 있는지 없는지 확인할 때 사용하는 키워드입니다.  
활용하는 방법은 다음과 같습니다.  
 
예제: 값 in 자료  
설명: 값이 자료에 있으면 True이고 없으면 False가 됩니다.  
  
예제:  


In [188]:
listdata = [1, 2, 3, 4]
a = 5 in listdata
b = 4 in listdata
print(a)               # False
print(b)               # True

False
True


https://rfriend.tistory.com/284 numpy 여러 문법

문제107. 모평균이 30이고 모표준편차가 7인 모집단을 구성하시오!

In [256]:
import numpy as np

avg = 30
std = 7
N = 1000000

mogipdan = np.random.randn(N) * std + avg  #randn : 가우시안 분포에서 난수생성
print(mogipdan)

[35.67235119 35.0786065  44.62372876 ... 30.37257007 27.9586741
 33.67063143]


설명: np.random.randn(숫자)를 쓰면 이 숫자만큼 가우시안 정규분포를 따르는 난수들이 숫자만큼 생성됨.  
문제108. 위의 모집단의 모평균을 출력하시오!


In [252]:
np.mean(mogipdan) # 모평균의 불편추정량

30.00730050781283

문제109. 아래의 SQL을 판다스로 구현하시오!  
>SQL  
select ename, job  
from emp  
where job in ('SALESMAN', 'ANALYST');

In [232]:
filename = '/content/emp3.csv'
import pandas as pd

emp = pd.read_csv(filename)
print(emp[['ename','job']][emp['job'].isin(['SALESMAN','ANALYST']) ]) # 어우 대괄호랑 소괄호 어지러워

     ename       job
4   MARTIN  SALESMAN
5    ALLEN  SALESMAN
6   TURNER  SALESMAN
8     WARD  SALESMAN
9     FORD   ANALYST
11   SCOTT   ANALYST


문제110. 아래의 SQL을 판다스로 구현하시오!  
SQL>  
select ename, job  
from emp  
where job not in ('SALESMAN','ANALYST');

In [237]:
print(emp[['ename','job']][~ emp['job'].isin(['SALESMAN','ANALYST'])]) #물결표시로 NOT을 구현한다.

     ename        job
0     KING  PRESIDENT
1    BLAKE    MANAGER
2    CLARK    MANAGER
3    JONES    MANAGER
7    JAMES      CLERK
10   SMITH      CLERK
12   ADAMS      CLERK
13  MILLER      CLERK


# 33. 문자열 이해하기

문자열은 문자나 기호 순서로 나열되어 있는 시퀀스 자료입니다.
- 문자열을 선언하는 방법 3가지
1.'문자열' <---------- 싱글 쿼테이션 마크  
2."문자열" <---------- 더블 쿼테이션 마크  
3."""문자열""" <------ 더블 쿼테이션 마크를 3개  
.  
문자에 아래와 같이 싱클 쿼테이션 마크(')가 포함되어져 있으면 위의 두번째인 더블 쿼테이션 마크를 사용하면 됩니다.  

In [238]:
a = "My son's name is john"
print(a)

My son's name is john


문자에 아래와 같이 더블 쿼테이션 마크가 있으면 위의 세번째인 """ """ 더블 3개를 사용하면 됩니다.

In [240]:
b = """My son's name is "john" """
print(b)

My son's name is "john" 


문제111. 아래와 같은 글씨가 출력되게 하시오!  
.  
"모집단"의 모평균은  
"모집단"의 모분산은  
"모집단"의 모표준편차는

In [244]:
print('"모집단"의 모평균은')
print(""""모집단"의 모분산은""")
print(""""모집단"의 모표준편차는""")

"모집단"의 모평균은
"모집단"의 모분산은
"모집단"의 모표준편차는


문제112. 모평균이 30이고 모표준편차가 7인 모집단 1000000개의 모평균과 모분산과 모표준편차를 아래와 같이 출력하시오.  
.  
"모집단"의 모평균은   xx.  
"모집단"의 모분산은   xx.  
"모집단"의 모표준편차는 xx.  

In [258]:
a = '"모집단"의 모평균은 '
b = '"모집단"의 모분산은 '
c = '"모집단"의 모표준편차는 '
print(a + str(np.mean(mogipdan)))
print(b + str(np.var(mogipdan)))
print(c + str(np.std(mogipdan)))

"모집단"의 모평균은 30.009619334157495
"모집단"의 모분산은 48.97478850025149
"모집단"의 모표준편차는 6.998198946889942


문제113. 모평균이 30이고 모표준편차가 7인 모집단(1000000개)에서 표본을 49개를 뽑으시오.

In [271]:
avg = 30
std = 7
N = 1000000

mogipdan = np.random.randn(N) * std + avg  #randn : 가우시안 분포에서 난수생성
print(np.random.choice(mogipdan,49))       # np. random. choice 추출 자유도까지 설정 가능


[37.81555715 31.89459198 45.73481879 31.2986874  40.81147611 34.15801619
 28.97478641 29.85167135 27.94558398 32.9217665  32.04403281 32.14641689
 29.75784267 29.98102123 13.67629072 29.4781279  32.08583584 20.22811415
 33.39544426 37.12326499 26.86704696 22.43125732 18.42542149 26.94569625
 26.04538374 23.41045576 38.23406492 27.18139495 20.64267226 36.18827874
 27.65024229 30.49101267 25.51011912 34.70624925 25.20642076 22.25712446
 23.89369863 16.63341667 24.69693083 33.87364748 19.42459917 24.13379822
 23.42566993 32.6121195  31.71580206 40.98130769 25.79455072 37.69544195
 30.93103406]


문제114. 위에서 뽑은 49개의 평균값을 출력하시오!

In [283]:
 print(  np.random.choice(mogipdan,49).mean()  ) # 더 간편한 mean

31.949884027892953


설명: numpy의 random 안의 choice 함수를 이용해서 mogipdan에서 49개를 표본샘플링하고 그 표본의 평균값을 출력한다.  
문제115. 위에서 출력한 표본의 평균값을 하나가 아니라 100개가 출력되게 하시오!

In [284]:
for i in range(1,101):
  print(np.random.choice(mogipdan).mean())

39.271112154388106
31.460296103072107
22.704674675638856
22.928033234117777
20.309321726636732
29.066442385612334
52.51828251366227
16.079324979955945
25.458859451724273
19.671760826776172
41.04574371511885
23.057412569426432
33.10442369451163
39.13360342211
25.01644889405498
20.61293066398433
24.823382577870102
34.01169077332292
38.57538227082394
20.463865997798436
31.3323444026922
26.86317389185539
25.47745669321036
28.4612450734247
31.92616767035915
41.86759613048636
35.27204280379423
23.818322513824356
22.76238456207409
32.70824918935811
35.37464315253989
34.84573510674841
39.794345067247306
23.801605733766927
25.004201860011655
15.141730040085216
22.96768447915186
32.99021160568842
29.111741282654503
29.678196610685017
26.731420700226582
30.013963552412342
26.427895513704794
26.49658667977736
34.70399005312177
29.10568751341549
34.64901807723223
36.37847135002743
24.40332359899208
30.57959487988927
24.135994168276945
31.656219184023094
22.16665274715433
37.969523472082
33.82528933

문제116. 이번에는 100개를 출력하지 말고 비어있는 리스트 a를 선언하고 a에 100개를 담으시오!

In [291]:
import numpy as np
avg = 30
std = 7
N = 1000000
mogipdan = np.random.randn(N)*std + avg

a = []
for i in range(1,101):
  a.append(np.random.choice(mogipdan,49).mean())
print(a)

[29.048143963686655, 29.77753095443178, 29.40902802718471, 31.838322763623324, 30.505273810336178, 29.96064860578475, 29.79255982312779, 30.418463808564656, 29.904512701739762, 29.021293122549228, 29.359602247060113, 30.09407260304218, 30.373346852203582, 28.546136864062554, 29.057145752349935, 29.239791135047824, 29.782819141500262, 27.37501547809361, 30.436099503643433, 30.540982705681216, 28.988607210330592, 27.414782607852125, 29.324327959343467, 28.736646652531423, 29.075230585347395, 30.38362553213882, 29.36869086677039, 29.89397776771812, 28.76776309506383, 27.7814993304253, 29.763501133242958, 30.051995019927276, 31.401869284063515, 29.787157372207506, 31.45120456266749, 30.668584891408393, 29.804890621569964, 30.410831544780198, 27.806820984183346, 30.402979713725568, 31.223133897553296, 29.336422840349737, 29.722413059261534, 28.864437864561967, 29.502159534951414, 31.147773072557932, 29.499468151443143, 29.51223323987485, 30.369706409305717, 30.950101111048415, 31.9751659833

문제117. 모평균이 30이고 모표준편차가 7인 모집단에서 임의추출한 크기가 49인 표본의 표본평균의 평균과 표준편차를 구하여라

In [336]:
import numpy as np
avg = 30
std = 7
N = 1000000
a = []

mogipdan = np.random.randn(N)*std + avg

for i in range(1,101):
  a.append(np.random.choice(mogipdan,49).mean())
print(np.mean(a))
print(np.var(a))
print(np.std(a))

30.013248562614553
1.1089141240144158
1.053049915253031


```
포멧 문자열  설명
%s           문자열에 대응됨
%c           문자나 기호 한개에 대응됨
%f           실수에 대응됨
%d           정수에 대응됨
%%           %라는 기호 자체를 표시함

예제:
```

In [326]:
txt1 = '자바'
txt2 = '파이썬'
num1 = 5
num2 = 10
print( '나는 %s 보다 %s에 더 익숙합니다' %(txt1, txt2) )

나는 자바 보다 파이썬에 더 익숙합니다


문제118. 아래의 변수를 이용해서 아래와 같이 결과가 출력되게 하시오!  
num1 = 5  
num2 = 10  
결과 : 5는 10보다 작습니다.

In [328]:
num1 = 5
num2 = 10
print('%d는 %d보다 작습니다' %(num1,num2))

5는 10보다 작습니다


문제119. 문제117번의 결과가 아래와 같이 출력되게 하시오!  
(문자열 포멧을 이용하세요)  
표본평균의 평균값은 29.918451681179658 이고 분산값은 1.1878482150313316 이고 표준편차는 1.0898844961881657입니다.

In [330]:
num1 = 29.918451681179658
num2 = 1.1878482150313316
num3 = 1.089884496188165
print("표본평균의 평균값은 %f이고 분산값은 %f이고 표준편차는 %f입니다." %(num1,num2,num3))

표본평균의 평균값은 29.918452이고 분산값은 1.187848이고 표준편차는 1.089884입니다.


문제120. 위의 결과가 아래와 같이 소수점 두번째 자리까지만 나오게 하시오!  
표준평균의 평균값은 29.92이고 분산값은 1.19이고 표준편차값은 1.09입니다.

In [345]:
print("표본평균의 평균값은 %.2f이고 분산값은 %.2f이고 표준편차는 %.2f입니다." %(round(np.mean(a),2),round(np.var(a),2),round(np.std(a),2))) # %.2f 하면 2자리에서 끊긴다.

표본평균의 평균값은 30.01이고 분산값은 1.11이고 표준편차는 1.05입니다.


문제121.(통계를 코드로 구현 문제 4번의 링크2)  
어느 비행기 탑승객의 짐의 무게는 평균이 18kg이고 표준편차가 3kg인 정규분포를 따른다고 한다. 이 비행기 탑승객 중에서 36명을 임의추출할 대, 짐의 평균 무게가 17kg 이상일 확률을 구하여라. (단, Z가 표준정규분포를 따르는 확률변수일 때, P(0<= Z <= 2) = 0.4772이다)

In [424]:
import numpy as np
# 짐의 무게 N ~ (18, 3^2)

n = 36
mu = 18
std = 3/np.sqrt(n)


Z = (17 - mu)/std   # Z = -2이고  P(0<= Z <= 2) = 0.4772 이므로
P = 0.5 + 0.4772
print(P)


0.9772000000000001
